discord/user.py

```diff
@@ -849,3 +849,28 @@ def transform(d):
                        mutual_guilds=mutual_guilds,
                        user=self,
                        connected_accounts=data['connected_accounts'])
+
+    async def set_note(self, note):
```

Comment:
Honestly not sure I like this name.

Let me disregard the fact that this is a user-bot only feature for a minute.

In this library whenever you want to edit something about the model you don't use any `set_` methods, you have an `edit` method with a kwarg-only interface. Now, `User.edit` doesn't make much sense to add for only this thing but it's what would be the most consistent. The problem is that the concept of editing a `User` instance and adding an `edit` to it is that it breaks expectations when using `Member.edit`. Why couldn't you edit the note on a Member with the same interface?

These design decisions are why personally I'm not sure this is a good fit. Maybe if the name was something else. Maybe it's unimportant.